Documentation updates

elizabethengelman opened the pull request:
A couple of things that I'm planning to add to the README/documentation, if they seem useful:
- [x] mention how reorgs are handled
- [x] clarify the relationship between which Ethereum client sync  mode can be used with each command
- [x] rename `sync` to `fullSync`
- [x] include a config file example for `sync`/`headersSync`

Other questions/ideas for improvements to the code base (that I would love some input on!):
1. is there a different name we want to discuss for the `Watcher` abstraction? 
1. allow for the validation window during a sync to be configurable **UPDATE: added a story in Jira, VDB-591**
1. based on the workflows we currently have, I don't think we're using the `coldImport` command - should we mention this explicitly somehow? Perhaps we need some more clarity about what it's utility is.
1. should we rename the `sync` command to `fullSync` since that is what the tables are called? I am worried about confusing this with geth's full sync though 🤔  **UPDATE: added to checklist above**
1. As I was revisiting it, I found it a bit confusing that `EventWatcher.execute` is adding transactions to the database. I may be overthinking this, but I can imaging that this may be confusing for a new user, who though they did (and understood) a `headerSync`, and then after executing transformers realize that they now have transactions in their db. I may be overthinking it, and this is just something that we need to explicitly document in the right place, and the right way. So I'm totally up for suggestions!

elizabethengelman commented on README.md:
```diff
@@ -126,9 +138,6 @@ Documentation on how to build custom transformers to work with these commands ca
-
 
 ## Contributing
 Contributions are welcome! For more on this, please see [here](../staging/documentation/contributing.md).
```
I would love to include a Code of Conduct in here, for when we receive community contributions. The [Contributor Covenant](https://www.contributor-covenant.org/version/1/4/code-of-conduct) is one that comes to mind, but I'm certainly happy to do some more research/work on finding one that fits Vulcanize.

elizabethengelman replied:
8dc9ddf65e4004bcb430892594f4cc999a0e6393

elizabethengelman commented on documentation/contributing.md:
```diff
@@ -1,11 +1,26 @@
+- Update the README or any documentation files as necessary. If editing the Readme, please
+conform to the
+[standard-readme specification](https://github.com/RichardLitt/standard-readme).
+- You may merge a Pull Request once you have an approval from core developer.
```
This is the general workflow that we've been following, but I wonder if we should rethink this in the future when additional contributors come onto the project so that we can continue to sanely add new features. Do folks think that we should require 2 approvals? Only allow "core developer" merging permissions?

elizabethengelman replied:
I wonder if we should also mention to tag a few of us as Reviewers on any new Pull Request?

rmulhol replied:
Definitely on board with requiring 2 approves and only allowing core members to merge PRs. I think (hope) the latter is already the case

elizabethengelman replied:
👍 cool, I'll update the language in this document, and change the repo settings to require 2 reviews before allowing a merge.

i-norden commented on documentation/sync.md:
```diff
@@ -1,22 +1,35 @@
 # Syncing commands
```
`step_one` or maybe something generic like `eth_syncing`? `step_one` lends itself in my eyes to having docs named as subsequent steps which might be tricky since the next step would be a split choice between the custom transformers or the generic transformer.

elizabethengelman replied:
Yep, I totally agree to a rename.

I agree that having it called `step_one` would make it feel like the rest would need to be named `step_x`.  If we did make that change, then we could rename `custom-transformers` and `generic-transformer` to `step-two-custom-form` and `step-two-generic-transform`, respectively?

elizabethengelman replied:
I renamed it `data-syncing` for now, though I'm not sure that's quite right either. 🤔 

829a581727e7d87e962a97485cd3e3718ae19e11

elizabethengelman commented on documentation/contributing.md:
```diff
@@ -1,11 +1,23 @@
 can be run together with other custom transformers using the [composeAndExeucte](../../staging/documentation/composeAndExecute.md) command.
 
+## Pull Requests
+- `go fmt` is run as part of `make test` and `make integrationtest`, please make sure to check in the format changes.
```
I think we're specifically using the `go fmt` package, as opposed to `gofmt` - I had no idea they were two different things! 🤯

i-norden replied:
Oh! This is news to me also. Whoops, I probably should have noticed that in our Makefile now by now 🙃

elizabethengelman commented on documentation/contributing.md:
```diff
@@ -1,11 +1,23 @@
+- Make sure the build is passing.
+- Update the README or any [documentation files](./) as necessary. If editing the Readme, please
+conform to the
+[standard-readme specification](https://github.com/RichardLitt/standard-readme).
```
Good call, I think in the standard readme spec it includes `Dependencies` as a subsection of `Install`. It also doesn't include a `Tests` section - which I think could make sense as a subsection to either `Install` or `Usage`. I'd be happy to make those changes  - what do you all think?

elizabethengelman commented on documentation/sync.md:
```diff
@@ -1,22 +1,35 @@
-1. Start Ethereum node
+Syncs block headers from a running Ethereum node into the VulcanizeDB table `headers`.
+- Queries the Ethereum node using RPC calls.
+- Validates headers from the last 15 blocks to ensure that data is up to date.
```
cool, just added a story to the back log: https://makerdao.atlassian.net/browse/VDB-591

elizabethengelman commented on documentation/custom-transformers.md:
```diff
@@ -1,43 +1,78 @@
+   * [Example 2](https://github.com/vulcanize/ens_transformers/tree/master/transformers/registry)
+   * [Example 3](https://github.com/vulcanize/ens_transformers/tree/master/transformers/resolver)
+
+Contract Transformers
```
👍

elizabethengelman commented on documentation/custom-transformers.md:
```diff
@@ -1,43 +1,78 @@
+    * If the base vDB migrations occupy this path as well, they need to be in their `goose fix`ed form
+    as they are [here](../../staging/db/migrations)
+
+To update a plugin repository with changes to the core vulcanizedb repository, run `dep ensure` to update its dependencies.
```
That would be great if you could take the lead on reworking this @i-norden, let me know if you need someone to bounce ideas off, I'd be happy to help!

elizabethengelman commented on documentation/custom-transformers.md:
```diff
@@ -1,43 +1,78 @@
+    environment, i.e. we need to `compose` and `execute` the plugin .so file with the same exact version of vulcanizeDB.
+    * The plugin migrations are run during the plugin's composition. As such, if `execute` is used to run a prebuilt .so
+    in a different environment than the one it was composed in then the migrations for that plugin will first need to
+    be manually ran against that environment's Postgres database.
```
Updated it in b3019c99950f264b2fbffc1d0b8c3d8af8dd6cb1, let me know what you think!

elizabethengelman commented on documentation/sync.md:
```diff
@@ -1,22 +1,35 @@
+- Useful when you want to maintain a broad cache of what's happening on the blockchain.
+
+##### Usage
+1. Start Ethereum node (**if fast syncing your Ethereum node, wait for initial sync to finish**).
```
👍 updated in 0cc93b78e48d4805069d3d751b437a2e43303c2a

elizabethengelman commented:
Thanks so much for the feedback @rmulhol  & @i-norden! I think I've addressed most of what you've mentioned. A couple of things that I didn't touch because I think we probably need to think on them as a team a bit more are:
* potentially a new name for the Watcher abstraction
* removing/moving `coldImport` and `fullSync` commands
* Does adding transactions in `EventWatcher.execute` make sense? Where else could this go?

I also removed the check list item for adding a diagram for the db schema for now. At the moment it seems like `blocks` and `headers` are functionally the most interesting tables for core vulcanize, and a diagram relating those two tables didn't really seem necessary. I'd be happy to revisit this later though!
